### DISCUSSION TITLE
The id of cells changes on importing graph with fromJSON

### DISCUSSION THREAD

**Discussion by Biowavix** [clientIO/joint#2035] [Q&A] [answered]
I am not sure why, but when I import the following graph with the graph.fromJSON function:

![image](https://user-images.githubusercontent.com/92020001/217598067-95d72212-92e9-4054-9062-1bcdc0e0e999.png)

It throws the following error:

```
LinkView.mjs:977 Uncaught Error: LinkView: invalid source cell.
    at child.updateEndProperties (LinkView.mjs:977:1)
    at child.confirmUpdate (LinkView.mjs:87:1)
    at child.updateView (Paper.mjs:715:1)
    at child.updateViewsBatch (Paper.mjs:909:1)
    at child.updateViewsAsync (Paper.mjs:799:1)
```
I check that line and I see that the ids of my cells changed, so when the link tries to find the source and the target, it fails. The following photo shows it:

![image](https://user-images.githubusercontent.com/92020001/217599587-b49934cc-1a1e-4a59-a2aa-76c28e8c1df4.png)

Any idea/suggestion where to start looking for this error?

Thank you in advance!

**Comment by kumilingus** [comment, marked as answer]
Possibly a duplicate of https://github.com/clientIO/joint/discussions/1841#discussioncomment-3877734.

**Reply by Biowavix** [reply]
It was that! Thank you and sorry for the duplicate
